File: solver.c
#include<stdio.h>
#include<stdlib.h>

void simplify(int *source, int *target, int count, int lvl){
	int t_factor = target[lvl];
	int s_factor = source[lvl];
	for(int i = lvl; i < count+1; i++)
		target[i] = source[i]*t_factor - target[i]*s_factor;
}
int main(){
	int count;
	printf("Count of unknowns: ");
	scanf("%d", &count);
	int matrix[count][count+1];
	printf("Matrix:\n");
	for(int i = 0; i < count; i++)
		for(int j = 0; j < count+1; j++)
			scanf("%d", &matrix[i][j]);

	int lvl = 0;
	int *source = matrix[lvl];
	int *target;
	while(lvl < count-1){
		for(int i = 1; i+lvl < count; i++){
			target = matrix[lvl+i];
			simplify(source, target, count, lvl);
		}
		source = matrix[++lvl];
	}

	for(int i = 0; i < count+1; i++)
		printf("%d ", matrix[count-1][i]);

	return 0;
}

File: solver.py
import math

def abs(list):
    for i in range(0, list.len()):
        list[i] = abs(list[i])

    return list


def gcd(list):
    list = abs(list)
    if list.len() == 1:
        return list[0]
    if list.len() == 2:
        return math.gcd(list[0], list[1])
    for result in range(int(min(list)), 0, -1):
        if sum(list%result) == 0:
            return result




def solve(matrix, count):
    lvl = 0
    while lvl < count-1:
        for i in range(lvl+1, count):
            matrix[i] = matrix[i]*matrix[lvl][lvl] - matrix[lvl]*matrix[i][lvl]
            matrix[i] /= gcd(matrix[i])

        lvl += 1

    return [matrix[count-1][count-1], matrix[count-1][count]]


print("Count of unknowns: ")
count = int(input())
print("Matrix:")
matrix = [[]]
for i in range(count):
    for j in range(count+1):
        matrix[i] += [int(input())]

    matrix += [[]]

print(solve(matrix, count))
